### PR TITLE
Upload code full flow upload app to kobiton apprepo

### PR DESCRIPTION
**Pull request by dttan98**
cc @duylam anh review code giup em nha
https://kobiton.atlassian.net/browse/KOB-5046

### CODE REVIEW

File: kobiton-rest-api/upload-app-to-kobiton-apps-repo/shell-script/upload-app-s3.sh [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+echo 'Step 4: Create Application Or Version'
+JSON="{\"filename\":\"${FILENAME}\",\"appPath\":\"${APPPATH}\"}"
+curl -X POST \
+'https://api.kobiton.com/v1/apps' \
+-H "Authorization: Basic ${credentials}" \
+-H 'content-type: application/json' \
+-d "{\"filename\":\"${FILENAME}\",\"appPath\":\"${APPPATH}\"}"
+
+echo '...Done'
```
Comment (khanhdodang) [inline kobiton-rest-api/upload-app-to-kobiton-apps-repo/shell-script/upload-app-s3.sh]: Missing newline

File: kobiton-rest-api/upload-app-to-kobiton-apps-repo/nodejs/upload-app-s3.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        'filename' : `${FILE_NAME}`,
+        'appPath': `${getUrl.appPath}`
+      },
+      headers: headers
+    }, function (err, response, body) {
+      if (err) {
+        console.error('Error:', err);
+        return reject(err)
+      }
+      //console.log('Response:', response);
```
Comment (khanhdodang) [inline kobiton-rest-api/upload-app-to-kobiton-apps-repo/nodejs/upload-app-s3.js]: Remove this line

File: kobiton-rest-api/upload-app-to-kobiton-apps-repo/java/upload-app-s3.java [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+    public static void sleep(int milliseconds){
+        try{
+            Thread.sleep(milliseconds);
+        } catch (Exception ex){
+            ex.printStackTrace();
+        }
+
+    }
+
+}
```
Comment (khanhdodang) [inline kobiton-rest-api/upload-app-to-kobiton-apps-repo/java/upload-app-s3.java]: Add newline

File: kobiton-rest-api/upload-app-to-kobiton-apps-repo/nodejs/upload-app-s3.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
+// yarn install
```
Comment (khanhdodang) [inline kobiton-rest-api/upload-app-to-kobiton-apps-repo/nodejs/upload-app-s3.js]: Do we need a project with yarn is installed?

File: kobiton-rest-api/upload-app-to-kobiton-apps-repo/nodejs/upload-app-s3.js [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+// yarn add request
+
+const request = require('request');
+const fs = require('fs');
+const btoa = require('btoa');
+const USERNAME =  process.env.USERNAME
+const API_KEY =  process.env.API_KEY
+const FILE_NAME =  process.env.FILE_NAME
+const YOUR_APPLICATION_PATH = process.env.YOUR_APPLICATION_PATH
```
Comment (khanhdodang) [inline kobiton-rest-api/upload-app-to-kobiton-apps-repo/nodejs/upload-app-s3.js]: This var doesn't make sense.

File: kobiton-rest-api/upload-app-to-kobiton-apps-repo/nodejs/upload-app-s3.js [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+        headers: headers
+      }, function (err, response, body) {
+        
+        if (err || response.statusCode != 200) {
+          console.log(err)
+          return reject(err);
+        
+        }
+        console.log('Response body:', body);
+        console.log('Uploading...')
```
Comment (khanhdodang) [inline kobiton-rest-api/upload-app-to-kobiton-apps-repo/nodejs/upload-app-s3.js]: Do we need `;`?

File: kobiton-rest-api/upload-app-to-kobiton-apps-repo/java/README.md [hunk 7]
```diff
@@ -0,0 +1,12 @@
+# Development
+
+## Setup environment
+### Java
+- Make sure java is install
```
Comment (HienNguyenT) [inline kobiton-rest-api/upload-app-to-kobiton-apps-repo/java/README.md]: was installed

File: kobiton-rest-api/upload-app-to-kobiton-apps-repo/java/main.java [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+
+        System.out.println("If you want to use your app at local, please comment the downloadFile() method");
+        Common.downloadFile("https://s3-ap-southeast-1.amazonaws.com/kobiton-devvn/apps-test/demo/com.dozuki.ifixit.apk", "com.dozuki.ifixit.apk");
+        System.out.println("Step 1: Generate Upload URL");
+
+        File file = new File(filePath);
+        filePath = file.getAbsolutePath();
+
+        String jsonData = Common.generateUploadURL(fileName);
+        JsonObject jsonObject =(JsonObject) (new JsonParser()).parse(jsonData);
```
Comment (HienNguyenT) [inline kobiton-rest-api/upload-app-to-kobiton-apps-repo/java/main.java]: jsonObject =(JsonObject)  = jsonObject = (JsonObject)

File: kobiton-rest-api/upload-app-to-kobiton-apps-repo/python/upload-app-s3.py [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+api_key = (USERNAME + ':' + API_KEY).encode('utf-8')
+print(api_key)
+auth = base64.b64encode(api_key).decode('utf-8')
+print(auth)
+headers = {
+    'Authorization': 'Basic ' + auth,
+    'Accept': 'application/json'
+}
+data = {
+    'filename' : FILE_NAME,
```
Comment (khanhdodang) [inline kobiton-rest-api/upload-app-to-kobiton-apps-repo/python/upload-app-s3.py]: `'filename' :` -> `'filename':`

File: kobiton-rest-api/upload-app-to-kobiton-apps-repo/shell-script/README.md [hunk 10]
```diff
@@ -0,0 +1,13 @@
+# Development
+
+## 1. Setup environment
+### Install ack
+- Make sure ack was install at: [link](https://www.elastic.co/guide/en/elasticsearch/reference/current/)getting-started-install.html)
```
Comment (HienNguyenT) [inline kobiton-rest-api/upload-app-to-kobiton-apps-repo/shell-script/README.md]: was installed

File: kobiton-rest-api/upload-app-to-kobiton-apps-repo/java/main.java [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+        String appURL = jsonObject.getAsJsonPrimitive("url").getAsString();
+
+        System.out.println("Step 2: Upload file to S3");
+
+        Common.uploadFileToS3(filePath, appURL);
+
+        System.out.println("Step 3: Create an app or app version");
+        String appResult = Common.createAnAppOrVersion(fileName, appPath);
+
+        jsonObject =(JsonObject) (new JsonParser()).parse(appResult);
```
Comment (khanhdodang) [inline kobiton-rest-api/upload-app-to-kobiton-apps-repo/java/main.java]: Check space

File: kobiton-rest-api/upload-app-to-kobiton-apps-repo/java/upload-app-s3.java [hunk 12]
```diff
@@ -0,0 +1,309 @@
...
+        return "";
+    }
+
+    public static String generateUploadURL(String filePath) {
+        return generateUploadURL(filePath, 0);
+    }
+    
+    public static String generateUploadURL(String filePath, int appId) {
+        try {
+
```
Comment (khanhdodang) [inline kobiton-rest-api/upload-app-to-kobiton-apps-repo/java/upload-app-s3.java]: Remove this line

File: kobiton-rest-api/upload-app-to-kobiton-apps-repo/java/upload-app-s3.java [hunk 13]
```diff
@@ -0,0 +1,309 @@
...
+        }
+    }
+
+    public static void sleep(int milliseconds){
+        try{
+            Thread.sleep(milliseconds);
+        } catch (Exception ex){
+            ex.printStackTrace();
+        }
+
```
Comment (khanhdodang) [inline kobiton-rest-api/upload-app-to-kobiton-apps-repo/java/upload-app-s3.java]: Remove this line

File: kobiton-rest-api/upload-app-to-kobiton-apps-repo/nodejs/.gitignore [hunk 14]
```diff
@@ -0,0 +1 @@
+node_modules
```
Comment (khanhdodang) [inline kobiton-rest-api/upload-app-to-kobiton-apps-repo/nodejs/.gitignore]: Add a newline

File: kobiton-rest-api/upload-app-to-kobiton-apps-repo/nodejs/package.json [hunk 15]
```diff
@@ -0,0 +1,16 @@
+{
+  "name": "nodejs",
+  "version": "1.0.0",
+  "description": "## 1. Setup environment ### Nodejs - Make sure nodejs version >= 8 is installed - yarn install - yarn add request ## 2 Coding - To run in local: `(environment variables) node upload-app-s3.js` * Set environment variables:     * USERNAME     * API_KEY     * FILE_NAME     * YOUR_APPLICATION_PATH - For example: `USERNAME=testuser API_KEY=123ed­123fac­9137dca FILE_NAME=test-app YOUR_APPLICATION_PATH=./Downloads/test-app.apk node upload-app-s3.js`",
```
Comment (khanhdodang) [inline kobiton-rest-api/upload-app-to-kobiton-apps-repo/nodejs/package.json]: This line is too long

File: kobiton-rest-api/upload-app-to-kobiton-apps-repo/shell-script/README.md (hunk 10 above)
Comment (khanhdodang) [reply kobiton-rest-api/upload-app-to-kobiton-apps-repo/shell-script/README.md]: @dttan98 pls address this comment

File: kobiton-rest-api/upload-app-to-kobiton-apps-repo/nodejs/upload-app-s3.js [hunk 16]
```diff
@@ -0,0 +1,95 @@
...
+            method: 'PUT',
+            url: `${getUrl.url}`,
+            headers: {
+              'Content-Length': stats.size,
+              'Content-Type': 'application/octet-stream',
+              'x-amz-tagging': 'unsaved=true'
+            }
+          },  
+          function (err, res, body) {
+              if(err){
```
Comment (khanhdodang) [inline kobiton-rest-api/upload-app-to-kobiton-apps-repo/nodejs/upload-app-s3.js]: Please make the script consistent, we can use 4 spaces or switch to 2 spaces

**Pull request merged**